File: need4spd/euler_53.pl
sub fact {
    my ($n) = @_;
    
    if ($n == 1) {
        return 1
    }
    
    my $result = 1;
    foreach my $d ((1..$n)) {
        $result = $result * $d;
    }
    
    return $result;
}

sub get_number_of_cases {
    my ($n, $r) = @_;
    
    my $result = fact($n) / (fact($r) * fact($n-$r));
    return $result;
}

my $cnt = 0;

foreach my $n ((23..100)) {
    foreach my $r ((1..$n-1)) {
        if (get_number_of_cases($n, $r) > 1000000) {
            $cnt += 1;
        }
    }
}

print "$cnt";

File: need4spd/euler_53.rb
def fact(n)
  if n<= 1
    1
  else
    n * fact( n - 1 )
  end
end

def get_number_of_cases(n, r)
  r = fact(n) / (fact(r) * fact(n-r))
end

cnt = 0
(23..100).each do |n|
  (1..n-1).each do |r|
    if get_number_of_cases(n, r) > 1000000
      cnt += 1
    end
  end
end

puts cnt
